cmd/up/controlplane/connect.go

```diff
@@ -0,0 +1,201 @@
+			return errors.Wrap(err, errReadParametersFile)
+		}
+		if err := c.File.Close(); err != nil {
+			return errors.Wrap(err, errReadParametersFile)
+		}
+	}
+	c.parser = helm.NewParser(base, c.Set)
+	return nil
+}
+
+// getCmd gets a single control plane in an account on Upbound.
+type connectCmd struct {
+	mgr     install.Manager
+	parser  install.ParameterParser
+	kClient kubernetes.Interface
+
+	Name  string `arg:"" required:"" help:"Name of control plane."`
+	Token string `required:"" help:"API token used to authenticate."`
+
+	Kubeconfig string `type:"existingfile" help:"Override default kubeconfig path."`
+	Namespace  string `short:"n" env:"MCP_CONNECTOR_NAMESPACE" default:"kube-system" help:"Kubernetes namespace for MCP Connector."`
+
+	install.CommonParams
+}
+
+// Run executes the get command.
+func (c *connectCmd) Run(p pterm.TextPrinter, upCtx *upbound.Context) error {
+	if err := c.applyMCPKubeconfig(upCtx); err != nil {
+		return err
+	}
+	params, err := c.parser.Parse()
+	if err != nil {
+		return errors.Wrap(err, errParseInstallParameters)
+	}
+	if err = c.mgr.Install("", params); err != nil {
+		return err
+	}
+
+	_, err = c.mgr.GetCurrentVersion()
+	if err != nil {
+		return err
+	}
+
+	p.Printfln("Connected to the MCP %q, ready for binding APIs!", c.Name)
+	return nil
+}
+
+func (c *connectCmd) applyMCPKubeconfig(upCtx *upbound.Context) error {
+	// Create a secret with MCP Kubeconfig
+	proxy := upCtx.ProxyEndpoint
+	id := path.Join(upCtx.Account, c.Name)
+	key := fmt.Sprintf(kube.UpboundKubeconfigKeyFmt, strings.ReplaceAll(id, "/", "-"))
+	proxy.Path = path.Join(proxy.Path, id, kube.UpboundK8sResource)
+
+	mcpCfg := &v1.Config{
+		Kind:       "Config",
+		APIVersion: "v1",
+		Clusters: []v1.NamedCluster{
+			{
+				Name: key,
+				Cluster: v1.Cluster{
+					Server: proxy.String(),
+				},
+			},
+		},
+		AuthInfos: []v1.NamedAuthInfo{
+			{
+				Name: key,
+				AuthInfo: v1.AuthInfo{
+					Token: c.Token,
+				},
+			},
+		},
+		Contexts: []v1.NamedContext{
+			{
+				Name: key,
+				Context: v1.Context{
+					Cluster:  key,
+					AuthInfo: key,
+				},
+			},
+		},
+		CurrentContext: key,
+	}
+
+	b, err := yaml.Marshal(mcpCfg)
+	if err != nil {
+		return errors.Wrap(err, "cannot marshal MCP kubeconfig")
+	}
+
+	secretName := "mcp-config"
+	versionV1 := "v1"
+	kindSecret := "Secret"
+	_, err = c.kClient.CoreV1().Secrets(c.Namespace).Apply(context.Background(), &applyconfigurationscorev1.SecretApplyConfiguration{
+		TypeMetaApplyConfiguration: applyconfigurationsmetav1.TypeMetaApplyConfiguration{
+			Kind:       &kindSecret,
+			APIVersion: &versionV1,
+		},
+		ObjectMetaApplyConfiguration: &applyconfigurationsmetav1.ObjectMetaApplyConfiguration{
+			Name:      &secretName,
+			Namespace: &c.Namespace,
+		},
+		Data: map[string][]byte{
+			"kubeconfig": b,
+		},
+	}, metav1.ApplyOptions{FieldManager: "up"})
+
+	if err != nil {
+		return errors.Wrap(err, "cannot apply MCP kubeconfig secret")
+	}
+
+	return nil
```

Comment:
nitpick: `errors.Wrap` returns `nil` if `err == nil`
```suggestion
	return errors.Wrap(err, "cannot apply MCP kubeconfig secret")
```